lib/rubocop/cop/minitest/useless_assertion.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Minitest
+      # Detects useless assertions (assertions that either always pass or always fail).
+      #
+      # @example
+      #   # bad
+      #   assert true
+      #   assert_equal @foo, @foo
+      #   assert_nil [foo, bar]
+      #
+      #   # good
+      #   assert something
+      #   assert_equal foo, bar
+      #   assert_nil foo
+      #   assert false, "My message"
+      #
+      class UselessAssertion < Base
+        MSG = 'Useless assertion detected.'
+
+        RESTRICT_ON_SEND = %i[
+          assert refute assert_nil refute_nil assert_not assert_empty refute_empty
+          assert_equal refute_equal assert_in_delta refute_in_delta assert_in_epsilon refute_in_epsilon
+          assert_same refute_same assert_includes refute_includes assert_silent
+        ].freeze
+
+        def on_send(node)
+          return if node.receiver
+
+          add_offense(node) if offense?(node)
+        end
+
+        private
+
+        # rubocop:disable Metrics
+        def offense?(node)
+          arg1, arg2, = node.arguments
+
+          case node.method_name
+          when :assert, :refute,
+               :assert_nil, :refute_nil,
+               :assert_not
+            arg2.nil? && arg1&.literal?
+          when :assert_empty, :refute_empty
+            return false unless arg1
+
+            arg1.str_type? || arg1.dstr_type? || arg1.array_type? || arg1.hash_type?
```

Comment:
This looks like it could use the same logic as above. So redundant? What do you think?

```suggestion
            arg2.nil? && arg1&.literal?
```

If so, the `when` array could be added to `SINGLE_ASSERTION_ARGUMENT_METHODS`.